### DISCUSSION TITLE
Support for multiple hosts

### DISCUSSION THREAD

**Discussion by chris-doehring** [spatie/ray#133] [Ideas]
I'm developing under a Windows environment with Docker. Usually it's completly enough to configure Ray to send anything to `host.docker.internal`, but sometimes I want to execute an artisan command without having to open a shell to the Docker container environment, but in that case Ray is not able to send anything since the target is configured as `host.docker.internal` and I don't want to change it since I only want to execute an artisan command and I would forget to change it back afterwards.

It would be really nice if Ray could support an array of hostnames where the same request will be send to. In my case I could set `host.docker.internal` and `localhost`, so it would send it twice, once to the correct address. In my opinion the port does not need to be configured multiple times since it will remain the same for all hostnames.

The amount of failed requests could be reduced with a check whether a given hostname resolves or not.

**Comment by AdrianMrn** [comment]
I don't have a lot of experience with programming on Windows and Docker, but am I correct in thinking a temporary solution could be to set the hostname to different environment variables on your machine and the docker container, and use that env variable in the config?

**Reply by chris-doehring** [reply]
Thanks for your answer! Since I share the same code base for both Windows and Docker this is not really possible, but it just came to my mind that a check in the `ray.php` validating the environment resolves that problem. It's also no longer an issue for me anyway, since I learned that docker automatically registers the hostname `host.docker.internal` in the Windows host environment. So the default docker config does the job, and if not one can extend the `ray.php` by some environment checks to differentiate between different hostnames.

**Reply by freekmurze** [reply]
Ok, we're leaving things as they are for now.
